core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPoint.java

```diff
@@ -38,6 +38,11 @@ public DefaultEndPoint(InetSocketAddress address) {
   @NonNull
   @Override
   public InetSocketAddress resolve() {
+    return retrieve();
```

Comment:
Nit: save yourself the method call overhead here and just return address directly... ?